### ISSUE TITLE
[torch.jit.script] Torchscript produces incorrect result when argmax result is used in indexing

### ISSUE THREAD

**Issue by illuminascent** [pytorch/pytorch#114543]
### 🐛 Describe the bug

Here is a minimum working example to reproduce the issue.
This function works on a 2-d array, when given a location (i, j), it iteratively mask out any value greater or equal to t[i, j] in the i-th row.

```
import torch

def test(t, i: int, j: int):
    while t[i, j] > 0:
        k = t[i].argmax()
        t[i, k].fill_(-1)
    return t

t = torch.arange(12).view([2, -1])
print(test(t, 0, 3))
test = torch.jit.script(test)
print(test.code)
t = torch.arange(12).view([2, -1])
# test(t, 0, 3)  # THIS HANGS THE PROCESS, DO NOT EXECUTE
```
If you look and the output, we can see the raw function gets correctly executed.
However, the compiled code is an infinite loop and the fill_ operation is nowhere to be found.
```
tensor([[ 0,  1,  2, -1, -1, -1],
        [ 6,  7,  8,  9, 10, 11]])
def test(t: Tensor,
    i: int,
    j: int) -> Tensor:
  _0 = torch.select(torch.select(t, 0, i), 0, j)
  _1 = bool(torch.gt(_0, 0))
  while _1:
    _2 = torch.select(torch.select(t, 0, i), 0, j)
    _1 = bool(torch.gt(_2, 0))
  return t
```

### Versions

PyTorch version: 2.1.0+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.25.0-rc2
Libc version: N/A

Python version: 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080
Nvidia driver version: 546.17
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture=9
CurrentClockSpeed=3792
DeviceID=CPU0
Family=198
L2CacheSize=2048
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=3792
Name=Intel(R) Core(TM) i7-10700K CPU @ 3.80GHz
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] torch==2.1.0+cu118
[pip3] torch-tb-profiler==0.4.3
[pip3] torchaudio==2.1.0+cu118
[pip3] torchvision==0.16.0+cu118
[conda] Could not collect

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
